frontend/src/core/comments/components/CommentsList.js

```diff
@@ -42,17 +42,35 @@ export default function CommentsList(props: Props) {
 
     const translationId = translation ? translation.pk : null;
 
+    const renderComment = (comment) => {
+        return (
+            <Comment
+                comment={comment}
+                canPin={canPin}
+                key={comment.id}
+                togglePinnedStatus={togglePinnedStatus}
+            />
+        );
+    };
+
+    const [pinnedComments, allComments] = comments.reduce(
+        (comment, currentComment) => {
+            comment[currentComment.pinned === true ? 0 : 1].push(
+                currentComment,
+            );
+            return comment;
+        },
+        [[], []],
+    );
```

Comment:
We should replace this block with something simpler like:
```js
const pinnedComments = comments.filter(comment => comment.pinned === true);
```

And then use `comments` instead of `allComments` below.

Reply:
I want to make sure I have this right before I push the code. I replace the block of code with the one you suggested. But know it doesn't Filter the `pinnedComments` from the original comments. It is now duplicating the comments instead of pinning the one to the top when the pin is selected. Do you want me to create a filter to filter the pinned comments from the original comments, or was this code you suggested to achieve this function?

Reply:
`.filter(comment => comment.pinned)` should suffice if I read that correctly.

Reply:
@mathjazz Just an FYI, I'm working with Mitch on this.

Reply:
Yup, comments should be duplicated. Pinned comments on top, then all comments in chronological order.

If you extract pinned comments, it's hard to follow the discussion.

Reply:
Got it! Do you want to remove the pin for all comments because when pinned it shows the pin in both places?

![Screen Shot 2021-01-21 at 10 15 15 AM](https://user-images.githubusercontent.com/69736237/105393784-9ae62a80-5bd1-11eb-9689-910665bc2e80.png)

Reply:
I think it's helpful to see the pin in the list of all comments as well, to get the sense of which comments are important when you're going through a list of comments, especially if it's long.